GitHubTeams.ps1

```diff
@@ -276,6 +315,488 @@ filter Get-GitHubTeamMember
+
+    .DESCRIPTION
+        Creates a team within an organization on GitHub.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OrganizationName
+        The name of the organization to create the team in.
+
+    .PARAMETER TeamName
+        The name of the team.
+
+    .PARAMETER Description
+        The description for the team.
+
+    .PARAMETER MaintainerName
+        A list of GitHub IDs for organization members who will become team maintainers.
+
+    .PARAMETER RepositoryFullName
+        The full name (e.g., "organization-name/repository-name") of repositories to add the team to.
+
+    .PARAMETER Privacy
+        The level of privacy this team should have.
+
+    .PARAMETER ParentTeamName
+        The name of a team to set as the parent team.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .INPUTS
+        System.String
+        GitHub.User
+
+    .OUTPUTS
+        GitHub.Team
+
+    .EXAMPLE
+        New-GitHubTeam -OrganizationName PowerShell -TeamName 'Developers'
+
+        Creates a new GitHub team called 'Developers' in the 'PowerShell' organization.
+
+    .EXAMPLE
+        $teamName = 'Team1'
+        $teamName | New-GitHubTeam -OrganizationName PowerShell
+
+        You can also pipe in a team name that was returned from a previous command.
+
+    .EXAMPLE
+        $users = Get-GitHubUsers -OrganizationName PowerShell
+        $users | New-GitHubTeam -OrganizationName PowerShell -TeamName 'Team1'
+
+        You can also pipe in a list of GitHub users that were returned from a previous command.
+#>
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSShouldProcess', '',
+        Justification = 'Methods called within here make use of PSShouldProcess, and the switch
+        is passed on to them inherently.')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification = 'One or more parameters (like NoStatus) are only referenced by helper
+        methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false
+    )]
+    [OutputType({$script:GitHubTeamTypeName})]
+    param
+    (
+        [Parameter(
+            Mandatory,
+            Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        [string] $OrganizationName,
```

Comment:
I think we should be able to pipe in a `GitHub.Organization` object here, meaning that this should also be `ValueFromPipelineByPropertyName`.

Reply:
Done